### PR TITLE
Fill in the profile form with values

### LINKED ISSUE #1996: Populate sections of main profile page
# Description
Fill in all the sections which display user data. Each one needs to check if it is missing required data, and if so show a text box and a link to the edit form.

This applies to Pools, About me, language, Gov info, Work Location and Work Preferences

[Wireframe](https://www.figma.com/proto/PrvXEDac0HVp7dF7SLWVFm/Candidate-profile-interfaces---V5.0.0?node-id=2340%3A4185&starting-point-node-id=2340%3A4185)

# Acceptance Criteria
- [ ] Includes Graphql operation to get user data.
- [ ] Fill in all the sections which display user data.
- [ ] Each one needs to check if it is missing required data, and if so show a text box and a link to the edit form.

### PR DESCRIPTION
**Pull request by vd1992**
resolves #1996 

Some issues, per wireframe, there isn't anything in the schema yet for "job duration" or "disliked operational requirements"
Intl-ing and localizing ignored until the form is appropriately filled out
GraphQL stuff works fine I think locally

### CODE REVIEW

**Comment by vd1992** [comment]
From local environment: 

image

**Comment by vd1992** [comment]
All needed fields pulled in now I think from Schema.
Functionality all present.

**Comment by vd1992** [comment]
You can use the Controls tool for Chromatic Storybook to pass in specific values if you have something in mind to test

**Review by petertgiles — CHANGES_REQUESTED** [review]
Whoah, this is looking cool!  :tada: A few suggestions:

There are some issues with the page when no information is there (for example, when not logged in):
- Many boxes are missing the "There are required fields missing" message.
- Field labels with no entered data should not be displayed.
It would be good to have a storybook story in this state to be able to review.

The naming of the "inner component" and "outer component" is a little out of convention, I think.  I suggest _ProfilePage_ for the outer component and _ProfileForm_ for the inner component to be more like _SearchPage_ and _SearchForm_.  I think keeping them in the same file like this with this name is fine though.

Do we need the giant _ProfilePageProps_ interface?  This is the _User_ type, isn't it?

Regarding _skills and experience_, James' PR should be in soon.  I think it's worth holding this one back until that's in so that you can complete that section.

**Review by petertgiles — CHANGES_REQUESTED** [review]
Please check all the enums, as well.  There are a few that are displaying as the raw enum instead of the localized description.

**Comment by vd1992** [comment]
> Whoah, this is looking cool! 🎉 A few suggestions:
> 
> There are some issues with the page when no information is there (for example, when not logged in):
> 
>     * Many boxes are missing the "There are required fields missing" message.
> 
>     * Field labels with no entered data should not be displayed.
>       It would be good to have a storybook story in this state to be able to review.
> 
> 
> The naming of the "inner component" and "outer component" is a little out of convention, I think. I suggest _ProfilePage_ for the outer component and _ProfileForm_ for the inner component to be more like _SearchPage_ and _SearchForm_. I think keeping them in the same file like this with this name is fine though.
> 
> Do we need the giant _ProfilePageProps_ interface? This is the _User_ type, isn't it?
> 
> Regarding _skills and experience_, James' PR should be in soon. I think it's worth holding this one back until that's in so that you can complete that section.

- Skill and Experiences is out of scope of the issue though
- Required boxes won't apply to Inclusion, and Government Info is missing that one field so that one will be pending. The other ones seem to have it appear. Not sure if they fail to appear in some cases though. 
- I don't know if I need the interface, so far I just try to avoid the red underlines through whatever means. 

Renamed components, empty fields hidden now, and added in Enum stuff. Standard Intl I'm leaving until I've made sure the page is complete with all the stuff I need to add.

**Comment by vd1992** [comment]
types tamed, type monstrosity eliminated

File: frontend/talentsearch/src/js/components/profile/ProfilePage/ProfilePage.tsx [hunk 1]
```diff
@@ -160,7 +269,25 @@ export const ProfilePage: React.FC<ProfilePageProps> = ({
                   description: "Title of the My hiring pools section",
                 })}
               </h2>
-              <p>Pool details</p>
+              <div
+                data-h2-bg-color="b(gray)"
+                data-h2-padding="b(all, m)"
+                data-h2-radius="b(s)"
+              >
+                {(candidateArray === null || candidateArray.length === 0) && (
```
Comment (petertgiles) [inline frontend/talentsearch/src/js/components/profile/ProfilePage/ProfilePage.tsx:277]: You're doing a lot of `=== null` and `=== undefined` and `=== 0` in this file. These are all considered falsy in javascript.  You could using the more concise form of direct boolean evalution like `!candidateArray || !candidateArray.length` to shorten the code.  I see you doing this in other places on this file so I suggest that form to be consisent.

File: frontend/talentsearch/src/js/components/profile/ProfilePage/ProfilePage.tsx [hunk 2]
```diff
@@ -249,7 +648,64 @@ export const ProfilePage: React.FC<ProfilePageProps> = ({
...
               </div>
-              <p>Government status details</p>
+              <div
+                data-h2-bg-color="b(gray)"
+                data-h2-padding="b(all, m)"
+                data-h2-radius="b(s)"
+              >
+                {/* !!!!!!!!!!!!
+                TODO
+                ADD GOVERNMENT EMPLOYEE TYPE TO THIS FORM FUNCTION, THEN ADD EMPLOYEE TYPE LOGIC HERE
```
Comment (petertgiles) [inline frontend/talentsearch/src/js/components/profile/ProfilePage/ProfilePage.tsx:658]: This is the missing field you mentioned?  Could you add the issue # here?

File: frontend/talentsearch/src/js/components/profile/ProfilePage/ProfilePage.tsx [hunk 3]
```diff
@@ -340,7 +920,74 @@ export const ProfilePage: React.FC<ProfilePageProps> = ({
...
+                        defaultMessage: "I identify as:",
+                        description:
+                          "Label preceding what groups the user identifies as part of, followed by a colon",
+                      })}{" "}
+                    </p>{" "}
+                    <ul data-h2-padding="b(left, l)">
+                      {isWoman && (
+                        <li data-h2-font-weight="b(700)">
+                          {intl.formatMessage({
+                            defaultMessage: "Woman",
```
Comment (petertgiles) [inline frontend/talentsearch/src/js/components/profile/ProfilePage/ProfilePage.tsx:957]: I feel like we use these EE groups everywhere even though they're not a graphql enum.  I suggest adding them to the localized constants file.

**Review by petertgiles — CHANGES_REQUESTED** [review]
Looking great!  I noticed a few more things: 

Checking null state:
- For _Diversity, equity and inclusion_ there should be a link for _employment equity groups_.
- Many of the boxes should have a _You haven't added any information here yet_ message.

For non null state:
- None of the stories have data for pool candidates.  Bug in component or query or seeder?
- For _Government information_ the items should be in an unordered list (dots)
- For the _About me_ section could you make the two columns collapse for small screens?
image

File: frontend/talentsearch/src/js/components/profile/ProfilePage/ProfilePage.tsx (hunk 2 above)
Comment (vd1992) [reply frontend/talentsearch/src/js/components/profile/ProfilePage/ProfilePage.tsx:658]: going to mention this component in #2329

**Comment by vd1992** [comment]
will edit comment this later

**Review by petertgiles — APPROVED** [review]
Nice work, Vachan.  It looks great!

**Comment by vd1992** [comment]
> Annoyingly pedantic comment: Should the component be called ProfileForm? It doesn't contain a form! ;)

Peter suggested it above, so I rolled with it!
"The naming of the "inner component" and "outer component" is a little out of convention, I think. I suggest ProfilePage for the outer component and ProfileForm for the inner component to be more like SearchPage and SearchForm. I think keeping them in the same file like this with this name is fine though."

**Pull request merged**